rust/rope/src/tree.rs

```diff
@@ -116,29 +123,48 @@ enum NodeVal<N: NodeInfo> {
 
 // also consider making Metric a newtype for usize, so type system can
 // help separate metrics
+
+/// A trait for quickly processing attributes of a NodeInfo.
+/// 
+/// For the conceptual background see the 
+/// [blog post, Rope science, part 2: metrics](https://github.com/google/xi-editor/blob/master/doc/rope_science/rope_science_02.md).
 pub trait Metric<N: NodeInfo> {
+    /// Return the number of boundarys in the NodeInfo::Leaf
+    /// 
+    /// TODO: What does the second argument do?
     fn measure(&N, usize) -> usize;
 
+    /// Return a offset in base units forwich self.from_base_units will return the argument.
```

Comment:
Maybe just: "Returns the offset, in base units,  for an offset in measured units."

I believe there can only a single possible result for a given input. (At least, I cannot think of an example where this is not the case.)

Reply:
I believe this is actually the largest, but I need to think about it more to make sure. I'd also prefer that this and `from_base_units` not be defined in terms of each other, but I do think it makes a ton of sense to explicitly write down invariants they need satisfy - `from_base_units(to_base_units(x)) = x` is one for valid `x`.